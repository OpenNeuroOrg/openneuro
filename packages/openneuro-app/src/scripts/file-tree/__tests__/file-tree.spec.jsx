import React from 'react'
import { mount } from 'enzyme'
import { ApolloProvider } from 'react-apollo'
import FileTree, {
  sortByFilename,
  sortByName,
  unescapePath,
} from '../file-tree.jsx'

// official Jest workaround for mocking methods not implemented in JSDOM
window.matchMedia =
  window.matchMedia ||
  function() {
    return {
      matches: false,
      addListener: function() {},
      removeListener: function() {},
    }
  }

/* eslint-disable */
jest.mock('react-spring', () => ({
  useSpring: jest.fn().mockImplementation(() => [{ mockProp: 1 }, jest.fn()]),
  animated: {
    path: () => <path data-testid="ANIMATED-COMPONENT" />,
    div: () => <div data-testid="ANIMATED-COMPONENT" />,
  },
}))
/* eslint-enable */

describe('FileTree component', () => {
  it('renders with default props', () => {
    expect(mount(<FileTree />)).toMatchSnapshot()
  })
  it('expands and closes when clicked', () => {
    // ApolloProvider isn't used in this test but must exist
    // When enzyme supports hooks, this can be simplified
    const wrapper = mount(
      <ApolloProvider client={{}}>
        <FileTree name="Top Level" />
      </ApolloProvider>,
    )
    expect(
      wrapper
        .find('button.btn-file-folder > i.type-icon')
        .hasClass('fa-folder-open'),
    ).toBe(false)
    wrapper.find('button').simulate('click')
    expect(
      wrapper
        .find('button.btn-file-folder > i.type-icon')
        .hasClass('fa-folder-open'),
    ).toBe(true)
    wrapper.find('button').simulate('click')
    expect(
      wrapper
        .find('button.btn-file-folder > i.type-icon')
        .hasClass('fa-folder-open'),
    ).toBe(false)
  })
  describe('sortByFilename()', () => {
    it('sorts the expected filename properties', () => {
      expect(
        sortByFilename(
          { name: 'abc', filename: 'xyz' },
          { name: 'xyz', filename: 'abc' },
        ),
      ).toBe(1)
    })
  })
  describe('sortByName()', () => {
    it('sorts the expected name properties', () => {
      expect(
        sortByName(
          { name: 'abc', filename: 'xyz' },
          { name: 'xyz', filename: 'abc' },
        ),
      ).toBe(-1)
    })
  })
  describe('unescapePath()', () => {
    it('does not alter an already escaped path', () => {
      expect(unescapePath('sub-01/anat')).toBe('sub-01/anat')
    })
    it('does unescapes any : characters', () => {
      expect(unescapePath('sub-01:anat')).toBe('sub-01/anat')
    })
    it('unescapes multiple : characters', () => {
      expect(unescapePath('sub-01:anat:image.nii.gz')).toBe(
        'sub-01/anat/image.nii.gz',
      )
    })
  })
})
