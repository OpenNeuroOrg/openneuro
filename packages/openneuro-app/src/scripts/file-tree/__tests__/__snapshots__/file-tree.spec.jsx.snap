// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileTree component renders with default props 1`] = `
<FileTree>
  <button
    className="btn-file-folder"
    onClick={[Function]}
  >
    <i
      className="type-icon fa fa-folder"
    />
     
    <i
      className="accordion-icon fa fa-caret-down"
    />
  </button>
</FileTree>
`;
